jubatus/core/nearest_neighbor/lsh_function.cpp

```diff
@@ -273,6 +357,59 @@ inline static void next_gaussian_float16(RND& g, float *out) {
 
 #endif  // #ifdef JUBATUS_USE_FMV
 
+inline static void init_cache(
+    uint32_t hash_num,
+    vector<float>& proj,
+    const cache_t& cache) {
+  if (cache.bool_test()) {
+    proj.reserve(hash_num);
+  }
+}
+
+inline static bool check_cache(
+    cache_t& cache,
+    uint32_t seed,
+    vector<float>& proj,
+    float v) {
+  if (cache.bool_test()) {
+    scoped_lock lk(cache->lock);
```

Comment:
I see that lru.get is not const function. So, this lock need to be mutex or write lock.